### PR TITLE
refactor(connector): [Noon] Remove Default Case Handling

### PR DESCRIPTION
**Pull request by SagarDevAchar**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
- Addresses Issue #2277 
- Modified `crates/router/src/connector/noon/transformers.rs`
  - Convert `NotImplemented` to `NotSupported` in default case

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
No test cases. As In this PR only error message have been populated for all default cases.

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [ ] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by swangi-kumari** [comment]
Hey @SagarDevAchar ,
The changes you made is right but you also have to fill the default cases for `WalletData` (line - 248) Please fill the matching arms for that too.

**Comment by SagarDevAchar** [comment]
@swangi-kumari got it!

**Comment by swangi-kumari** [comment]
Hey @SagarDevAchar  ,

Pls address the CI fails

Run `cargo +nightly` fmt for formatting.

and `cargo run --features openapi -- generate-openapi-spec` for validating generated openApi spec file. And commit it.

**Comment by SagarDevAchar** [comment]
@swangi-kumari 

> Pls address the CI fails

Will do!

> and `cargo run --features openapi -- generate-openapi-spec` for validating generated openApi spec file. And commit it.

I cannot seem to get `cargo run --features openapi -- generate-openapi-spec` or `cargo clippy` to run successfully
I am stuck with errors like:

```console
error[E0432]: unresolved import `core::any::Demand`
.
.
error[E0432]: unresolved imports `core::any::Demand`, `core::any::Provider`
.
.
error[E0425]: cannot find function `request_ref` in module `any`
.
.
```

How do I resolve this?
Is there a different version / build of rust I need to be using? (My `rustc` version is `1.75.0-nightly`)

**Comment by SanchithHegde** [comment]
@SagarDevAchar You don't necessarily need the nightly toolchain for compiling our code, please use the stable toolchain for the purpose. For formatting code alone, you can use the nightly toolchain using the command `cargo +nightly fmt`.

And I checked the changes in this PR, I don't think you need to specify the generics for the return type (`Ok::<NoonPaymentData, E>(...)`), only using `Ok(...)` should suffice.

**Comment by SagarDevAchar** [comment]
@SanchithHegde 

> @SagarDevAchar You don't necessarily need the nightly toolchain for compiling our code, please use the stable toolchain for the purpose. For formatting code alone, you can use the nightly toolchain using the command `cargo +nightly fmt`.

Got it!

> And I checked the changes in this PR, I don't think you need to specify the generics for the return type (`Ok::<NoonPaymentData, E>(...)`), only using `Ok(...)` should suffice.

Well, on router compilation with just `Ok(...)`, the toolchain gave out errors stating the need for type annotations for `Result`

image

Hence, I have added the type annotations to `Ok(...)` as `Ok::<NoonPaymentData, errors::ConnectorError>(...)` and the CI Checks succeed. Please review and let me know of any changes necessary!

**Review by ArjunKarthik — APPROVED** [review]

File: crates/router/src/connector/noon/transformers.rs [hunk 1]
```diff
@@ -199,13 +199,17 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for NoonPaymentsRequest {
...
+                        Ok::<NoonPaymentData, errors::ConnectorError>(NoonPaymentData::Card(
+                            NoonCard {
+                                name_on_card: req_card.card_holder_name.clone(),
+                                number_plain: req_card.card_number.clone(),
+                                expiry_month: req_card.card_exp_month.clone(),
+                                expiry_year: req_card.get_expiry_year_4_digit(),
+                                cvv: req_card.card_cvc,
+                            },
+                        ))
+                    }
```
[the remark this answers is not in the document]
Comment (SagarDevAchar) [reply crates/router/src/connector/noon/transformers.rs]: Thanks @SamraatBansal!

**Review by swangi-kumari — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @SagarDevAchar!

**Review by SamraatBansal — APPROVED** [review]

**Comment by deepanshu-iiitu** [comment]
Hey @SagarDevAchar  ,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.  
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.  
Thanks!

**Pull request merged**